src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -334,6 +334,76 @@ public async Task ApproveReviewAsync(ClaimsPrincipal user, string reviewId, stri
             await ToggleReviewApproval(user, review, notes);
         }
 
+        /// <summary>
+        /// Request namespace review for TypeSpec and mark related SDK language reviews as namespace approval requested
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="reviewId"></param>
+        /// <param name="notes"></param>
+        /// <returns></returns>
+        public async Task<ReviewListItemModel> RequestNamespaceReviewAsync(ClaimsPrincipal user, string reviewId, string notes = "")
+        {
+            ReviewListItemModel typeSpecReview = await _reviewsRepository.GetReviewAsync(reviewId);
+            
+            // Only allow for TypeSpec reviews
+            if (typeSpecReview.Language != "TypeSpec")
```

Comment:
Is namespace approval UI visible to all languages or only for TypeSpec?

Reply:
I am wondering whether we should restrict here or make it flexible so a user can request namespace approval even from other languages and server logic will find all related reviews and mark them as approval requested.